src/helpers.js

```diff
@@ -34,4 +34,54 @@ function getSetImmediate() {
   }
 }
 
-export {getDocument, newMutationObserver, getSetImmediate}
+let originalSetTimeout = global.setTimeout
+
+if (typeof window !== 'undefined') {
+  originalSetTimeout = window.setTimeout
+}
+
+let originalClearTimeout = global.clearTimeout
+
+if (typeof window !== 'undefined') {
+  originalClearTimeout = window.clearTimeout
+}
+
+/*
+ * Return the original setTimeout function so that all functions work as expected by a user
+ * if he/she utilizes dom-testing-library in a test where jest.fakeTimers() is used.
+ *
+ * global.useFakeTimers is used by us to make sure that we can still utilize fakeTimers in our tests.
+ *
+ * see: https://github.com/testing-library/dom-testing-library/issues/300
+ */
+function getSetTimeout(windowObject) {
```

Comment:
I've injected the windowObject as a dependency here because (as far as I know) there's no way in Jest to set the window object to be undefined. This seemed to be the best way to make this code testable.

Reply:
Instead of this, you could add a test for this case to https://github.com/testing-library/dom-testing-library/tree/master/src/__node_tests__

Other than that, this looks pretty good to me.

Reply:
@kentcdodds Do you mean instead of dependency injection? 

The node tests are still run within Jest and the main issue I faced was that in Jest window === globals (see: https://github.com/facebook/jest/issues/3692#issuecomment-408132529) and I can't set globals (and thus window) to undefined.

Reply:
Is the a reason the code can't be:

```js
return typeof window === 'undefined' ? global.setTimeout : window.setTimeout
```

Reply:
Updated!

Reply:
What I was saying is you no longer need the `windowObj` argument with the code I provided.

Reply:
Hmm, I get that but how would I be able to test that: "returns original getSetTimeout from global if window is undefined" without passing windowObject?

Perhaps we could chat about what the best approach to test that case would be?

Reply:
> returns original getSetTimeout from global if window is undefined

You can add another test to the node environment tests, as I said earlier:

> Instead of this, you could add a test for this case to https://github.com/testing-library/dom-testing-library/tree/master/src/__node_tests

Reply:
Ahhhhh, it finally clicked this morning. The node tests are run without JSDOM and thus without a window object. I'm new to the project, that wasn't clear to me.

I've updated the PR, removed the dependency injection of window and updated the tests for the node environment. Let me know what you think!